docs/getting-started/github-apps.rst

```diff
@@ -45,47 +44,117 @@ sync with the linked GitHub repository.
+    **Each workflow listed must have a unique (or no) name.**
 subclass
-    The descriptor language used for the workflow. Supported values include CWL, WDL, and NFL.
+    The descriptor language used for the workflow. Supported values include CWL, WDL, NFL (Nextflow), and GXFORMAT2 (Galaxy).
 primaryDescriptorPath
-    The path to the primary descriptor file in the Git repository
-testParameterFiles
-    An array of paths to test parameter files in the Git repository.
+    The absolute path to the primary descriptor file in the Git repository
+testParameterFiles (optional)
+    An array of absolute paths to test parameter files in the Git repository.
+
+Ex. .dockstore.yml with a single workflow
+
+.. code:: yaml
+
+   version: 1.2
+   workflows:
+      - subclass: CWL
+        primaryDescriptorPath: /Dockstore.cwl
+        testParameterFiles:
+            - /test/dockstore.cwl.json
+
+The above ``.dockstore.yml`` is for a single workflow. Note that the name is not present since it is optional.
+
+Ex. .dockstore.yml with multiple workflows
+
+.. important:: The **name** field is an optional field used when a repository has multiple workflows in it that a user wants to register
+    as separate entries on Dockstore. Each entry within a ``.dockstore.yml`` file corresponds to a unique entry on Dockstore.
+
+.. code:: yaml
+
+   version: 1.2
+   workflows:
+      - name: globalAligner
+        subclass: CWL
+        primaryDescriptorPath: /runGlobalAligner.cwl
+        testParameterFiles:
+            - /test/globalAligner.cwl.json
+      - name: localAligner
+        subclass: CWL
+        primaryDescriptorPath: /runLocalAligner.cwl
+        testParameterFiles:
+            - /test/localAligner.cwl.json
+
+A common pattern seen on Dockstore is GitHub repositories that store many workflows. The above ``.dockstore.yml``
+has two entries for workflows. Notice that each entry uses a different name. Names are required if you want 
+multiple workflows registered on Dockstore from a single GitHub repository. The names must be unique between
+entries of the `workflows` array. For each unique name present, an entry will be created on Dockstore.
 
 Service YML File
 +++++++++++++++++
 TODO
 
+Migrating Existing Workflows to use GitHub Apps
+-------------------------------------------------
+If you have already registered GitHub workflows on Dockstore using our old method which required you to refresh, you are still able to use GitHub Apps!
+The migration process is fairly straightforward. First, make sure to install our GitHub App onto the repository for the workflow that you want to migrate.
+Next, create a ``/.dockstore.yml`` file and push it to a branch on your GitHub repo, making sure that the name field matches the existing workflowname.
+If your workflow did not have a workflowname set, simply exclude the field. It is very important that these names match, or else Dockstore won't know which
```

Comment:
So a no-name has to match with no-name? Building on previous question, what if multiple workflows without extra names were manually registered from the same repo? When you migrate and have to add a name will it cause a duplication?

Reply:
Yes, a no name must name with a no name. This is because no-name is a specific entry on dockstore, whereas having a name (Ex. Foobar) is a different entry but from the same repository.

It is impossible for multiple workflows without extra names to be registered from the same repo. The paths (ex. github.com/org/foobar/name) must be unique, and if two workflows have the same repo but no name set, the paths are the same.

When you migrate, you must match the existing settings. So if you are migration a workflow with no workflow name set, you would have an entry in the dockstore.yml with no name set, and those map to each other. If you had a dockstore.yml with a name set, then the migration would not happen, since they are not referencing the same workflows. Instead a new workflow with that name will be created.

Reply:
Good to know, I've always added the extra-name so never ran into this. 
Do we just handle this as a blank extra-name or a null value? Cause I wonder what happens when the initial workflow was registered without an extra name, but then later you added another workflow with a specified extra-name (which you said is possible with manual register). Will the migration will still work? 
Sorry just trying to be thorough, and the doc language implies that it wouldn't. If it works then I don't think you need to add those specifics to the doc, but just trying to avoid potential bugs.

Reply:
I think it is null. With the case you mentioned, they are treated as separate workflows, so the migration will still work.